### PR TITLE
AEROGEAR-2905 - Ignore cases when looking up configurations based on …

### PR DESCRIPTION
**Pull request by rachael-oregan**
…types

## Motivation

JIRA: https://issues.jboss.org/browse/AEROGEAR-2905

GitHub Issue: #246 

## Description

Refactored service configuration *type* lookup to ignore the letter case requested and search for lowercase 

## Progress

- [x] refactor service config *type* lookup to ignore the letter case used in the request

### CODE REVIEW

**Comment by ziccardi** [comment]
@rachael-oregan : I think we should add a 'toLowercase' [here](https://github.com/aerogear/aerogear-android-sdk/blob/master/core/src/main/java/org/aerogear/mobile/core/configuration/MobileCoreConfiguration.java#L32) as well so that we are sure that map keys are lowercase.

**Comment by rachael-oregan** [comment]
@ziccardi the search doesn't really matter here because `serviceConfigsByType.get(serviceType);` always returns null as `serviceConfigsByType` is just initialised 3 lines before doing the search. I'm not even sure why it's checking for null here `if (configsForType == null)` because `configsForType` is always going to be null

**Comment by ziccardi** [comment]
@rachael-oregan : it is inside a loop. And inside the loop we do:
`
serviceConfigsByType.put(serviceType, configsForType);
`
so if more than one configuration exists for a given type, we could get a non null value.

**Comment by rachael-oregan** [comment]
@ziccardi ah right ya ok, I'll update that now. Thanks 👍

**Comment by rachael-oregan** [comment]
@ziccardi mind having another look

**Comment by rachael-oregan** [comment]
@secondsun ya sure that makes sense, I'll add in some tests

File: core/src/main/java/org/aerogear/mobile/core/MobileCore.java [hunk 1]
```diff
@@ -246,7 +246,7 @@ public HttpServiceModule getHttpLayer() {
      * @return the configurations for this service type from the JSON config file
      */
     public List<ServiceConfiguration> getServiceConfigurationsByType(final String type) {
-        return serviceConfigsByType.get(type);
+        return serviceConfigsByType.get(type.toLowerCase());
```
Comment (wojta) [inline core/src/main/java/org/aerogear/mobile/core/MobileCore.java]: maybe sanity check or non null check would be good in this case

File: core/src/main/java/org/aerogear/mobile/core/MobileCore.java [hunk 2]
```diff
@@ -258,7 +258,7 @@ public HttpServiceModule getHttpLayer() {
      * @return the first configuration for this service type from the JSON config file
      */
     public ServiceConfiguration getServiceConfigurationByType(final String type) {
-        final List<ServiceConfiguration> configs = serviceConfigsByType.get(type);
+        final List<ServiceConfiguration> configs = serviceConfigsByType.get(type.toLowerCase());
```
Comment (wojta) [inline core/src/main/java/org/aerogear/mobile/core/MobileCore.java]: maybe sanity check or non null check would be good in this case

**Comment by rachael-oregan** [comment]
@secondsun @wojta mind having another look please

**Pull request merged**